tools/generate_definitions.py

```diff
@@ -0,0 +1,368 @@
+"""Script to generate the definitions.json file from rippled source code."""
+
+import os
+import re
+import sys
+
+import httpx
+
+if len(sys.argv) != 2 and len(sys.argv) != 3:
+    print("Usage: python " + sys.argv[0] + " path/to/rippled [path/to/pipe/to]")
+    print(
+        "Usage: python "
+        + sys.argv[0]
+        + " github.com/user/rippled/tree/feature-branch [path/to/pipe/to]"
+    )
+    sys.exit(1)
+
+########################################################################
+#  Get all necessary files from rippled
+########################################################################
+
+
+def _read_file_from_github(repo: str, filename: str) -> str:
+    url = repo.replace("github.com", "raw.githubusercontent.com")
+    url = url.replace("tree", "refs/heads")
+    url += filename
+    if not url.startswith("http"):
+        url = "https://" + url
+    response = httpx.get(url)
+    return response.text
+
+
+def _read_file(folder: str, filename: str) -> str:
+    with open(folder + filename, "r") as f:
+        return f.read()
+
+
+func = _read_file_from_github if "github.com" in sys.argv[1] else _read_file
+
+sfield_h = func(sys.argv[1], "/include/xrpl/protocol/SField.h")
+sfield_macro_file = func(sys.argv[1], "/include/xrpl/protocol/detail/sfields.macro")
+ledger_entries_file = func(
+    sys.argv[1], "/include/xrpl/protocol/detail/ledger_entries.macro"
+)
+ter_h = func(sys.argv[1], "/include/xrpl/protocol/TER.h")
+transactions_file = func(
+    sys.argv[1], "/include/xrpl/protocol/detail/transactions.macro"
+)
+
+
+# Translate from rippled string format to what the binary codecs expect
+def _translate(inp: str) -> str:
+    if re.match(r"^UINT", inp):
+        if re.search(r"256|160|128|192", inp):
+            return inp.replace("UINT", "Hash")
+        else:
+            return inp.replace("UINT", "UInt")
+
+    non_standard_renames = {
+        "OBJECT": "STObject",
+        "ARRAY": "STArray",
+        "AMM": "AMM",
```

Comment:
Since the left-side is identical to the right-side `AMM`, does this item need to be present in this map ?

Reply:
Yes, because otherwise it will be `Amm`. But this function is only used for STypes, so it's unnecessary as a result.